Run-benchmarks runs fast when run remotely

sundiamonde opened the pull request:
#### 09d15752c8fa09ebe355c3c1c3d8830b7ea359ec
<pre>
Run-benchmarks runs fast when run remotely
<a href="https://bugs.webkit.org/show_bug.cgi?id=243977">https://bugs.webkit.org/show_bug.cgi?id=243977</a>

Reviewed by Dewei Zhu.

Partial fix.  Running the test with non-default Safari will still have the same issue
To support non default builds  we launch Safari via the binary.
This means Safari inherits the session env instead of the full login env.
We can workarount this problem by using open to escape the ssh context.

Instead of using caffeinate to launch Safari set a timeout for it that matches the test timeout.

* Tools/Scripts/webkitpy/benchmark_runner/browser_driver/browser_driver.py:
(BrowserDriver.take_sleep_assertion):
* Tools/Scripts/webkitpy/benchmark_runner/browser_driver/osx_browser_driver.py:
(OSXBrowserDriver._launch_process):
(OSXBrowserDriver._terminate_processes):
(OSXBrowserDriver.take_sleep_assertion):
(OSXBrowserDriver._launch_process_with_caffeinate): Deleted.
* Tools/Scripts/webkitpy/benchmark_runner/browser_driver/osx_safari_driver.py:
(OSXSafariDriver.launch_url):
(OSXSafariDriver):
(OSXSafariDriver._launch_url_with_custom_path):
* Tools/Scripts/webkitpy/benchmark_runner/webserver_benchmark_runner.py:
(WebServerBenchmarkRunner):

Canonical link: <a href="https://commits.webkit.org/253495@main">https://commits.webkit.org/253495@main</a>
</pre>

dewei-zhu commented on Tools/Scripts/webkitpy/benchmark_runner/browser_driver/browser_driver.py:
```diff
@@ -43,6 +43,9 @@ def restore_env_after_all_testing(self):
     def diagnose_test_failure(self, debug_directory, error):
         pass
 
+    def take_sleep_assertion(self, timeout):
```
This can be a class method. Maybe we can make this a context, so that we can ensure the process is killed instead of invoking `subprocess.call(["pkill", "-9", "caffeinate"])`.
In the base class, it can be a empty context, which just does `yield`

dewei-zhu commented on Tools/Scripts/webkitpy/benchmark_runner/browser_driver/osx_browser_driver.py:
```diff
@@ -101,17 +105,10 @@ def _terminate_processes(cls, process_name, bundle_id):
-
-        subprocess.Popen(["/usr/bin/caffeinate", "-disw", str(process.pid)])
-        return process
+    def take_sleep_assertion(self, timeout):
```
In this class, we can do
```
caffeinate = subprocess.Popen(["/usr/bin/caffeinate", "-dist", str(timeout)])
yield
caffeinate.terminate()
```

dewei-zhu commented on Tools/Scripts/webkitpy/benchmark_runner/webserver_benchmark_runner.py:
```diff
@@ -51,8 +51,10 @@ def _run_one_test(self, web_root, test_file):
+            timeout = self._plan['timeout']
             self._browser_driver.launch_url(url, self._plan['options'], self._build_dir, self._browser_path)
-            with Timeout(self._plan['timeout']):
+            self._browser_driver.take_sleep_assertion(timeout)
```
Here, we can just do
```
with Timeout(timeout), caffeinate(timeout):
    pass
````

dewei-zhu commented on Tools/Scripts/webkitpy/benchmark_runner/browser_driver/osx_safari_driver.py:
```diff
@@ -23,12 +23,27 @@ def prepare_env(self, config):
 
     def launch_url(self, url, options, browser_build_path, browser_path):
+        if browser_build_path or browser_path:
+            self._launch_url_with_custom_path(self, url, options, browser_build_path, browser_path)
```
should we have a `return` here?

dewei-zhu reviewed (approved):
r=me with comments.

sundiamonde committed:
Run-benchmarks runs fast when run remotely
https://bugs.webkit.org/show_bug.cgi?id=243977

Reviewed by Dewei Zhu.

Partial fix.  Running the test with non-default Safari will still have the same issue
To support non default builds  we launch Safari via the binary.
This means Safari inherits the session env instead of the full login env.
We can workarount this problem by using open to escape the ssh context.

Instead of using caffeinate to launch Safari set a timeout for it that matches the test timeout.

* Tools/Scripts/webkitpy/benchmark_runner/browser_driver/browser_driver.py:
(BrowserDriver.take_sleep_assertion):
* Tools/Scripts/webkitpy/benchmark_runner/browser_driver/osx_browser_driver.py:
(OSXBrowserDriver._launch_process):
(OSXBrowserDriver._terminate_processes):
(OSXBrowserDriver.take_sleep_assertion):
(OSXBrowserDriver._launch_process_with_caffeinate): Deleted.
* Tools/Scripts/webkitpy/benchmark_runner/browser_driver/osx_safari_driver.py:
(OSXSafariDriver.launch_url):
(OSXSafariDriver):
(OSXSafariDriver._launch_url_with_custom_path):
* Tools/Scripts/webkitpy/benchmark_runner/webserver_benchmark_runner.py:
(WebServerBenchmarkRunner):

Canonical link: https://commits.webkit.org/253495@main
